From beb7a196721629e966fefc45d2fe85787f5211da Mon Sep 17 00:00:00 2001
From: Edmunt Pienkowsky <roed@onet.eu>
Date: Mon, 20 Dec 2021 10:27:22 +0100
Subject: [PATCH 1/2] LaCrosse TX29-IT: Increase pulse width to 58us


diff --git a/src/devices/lacrosse_tx35.c b/src/devices/lacrosse_tx35.c
index 8382c7c5..1573b36c 100644
--- a/src/devices/lacrosse_tx35.c
+++ b/src/devices/lacrosse_tx35.c
@@ -186,8 +186,8 @@ static char *output_fields[] = {
 r_device lacrosse_tx29 = {
         .name        = "LaCrosse TX29IT, TFA Dostmann 30.3159.IT Temperature sensor",
         .modulation  = FSK_PULSE_PCM,
-        .short_width = 55, // 58 us for TX34-IT
-        .long_width  = 55, // 58 us for TX34-IT
+        .short_width = 58,
+        .long_width  = 58,
         .reset_limit = 4000,
         .decode_fn   = &lacrossetx29_callback,
         .fields      = output_fields,
